cmd/ctr/commands/snapshots/snapshots.go

```diff
@@ -177,26 +167,51 @@ var diffCommand = cli.Command{
 	},
 }
 
-func getMounts(ctx gocontext.Context, id string, sn snapshots.Snapshotter) ([]mount.Mount, error) {
-	var mounts []mount.Mount
-	info, err := sn.Stat(ctx, id)
-	if err != nil {
-		return nil, err
-	}
-	if info.Kind == snapshots.KindActive {
-		mounts, err = sn.Mounts(ctx, id)
+func createDiff(ctx context.Context, aSnapshotID, bSnapshotID string, sn snapshots.Snapshotter, d diff.Comparer, opts ...diff.Opt) (ocispec.Descriptor, error) {
+	getMounts := func(id string) ([]mount.Mount, string, error) {
```

Comment:
How do you feel about func similar to `WithTempMount` and use callback instead? This way a caller doesn't have to manage `viewKey`.

```go
withMounts(ctx, idA, snap, func(aMounts) {
    return withMounts(ctx, idA, snap, func(bMounts) {
       return d.Compare(ctx, aMounts, bMounts, opts...)
    })
})
```

Reply:
This is a good idea, I am trying